src/i_video.c

```diff
@@ -841,7 +842,14 @@ void I_FinishUpdate(void)
                 break;
             }
 
-            if (target_time - elapsed_time > 1000)
+            uint64_t remaining_time = target_time - elapsed_time;
+
+            if (remaining_time > 250 && current_time - last_pump > 250)
+            {
+                last_pump = current_time;
+                SDL_PumpEvents();
```

Comment:
In my testing `SDL_PumpEvents` often takes longer than 100 us, occasionally takes longer than 200 us, and seldom takes longer than 250 us.